lib/Sub/Override.pm

```diff
@@ -80,8 +85,11 @@ sub wrap {
     {
         no strict 'refs';
         $self->{$sub_to_replace} ||= *$sub_to_replace{CODE};
+        my $code =  sub { $new_sub->( $self->{$sub_to_replace}, @_ ) };
+        my $prototype = prototype($new_sub);
```

Comment:
I can't see a downside to using the prototype from $sub_to_replace and stuffing that into the override sub. That way you don't have to specify the prototype on the new sub, it will 'just work'. Are there any concerns with that?

Reply:
There is no downside it is simply correct to do it that way :-)